aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -114,6 +115,33 @@
    */
   public static final String S3FILEIO_ACL = "s3fileio.acl";
 
+  /**
+   * If {@link org.apache.iceberg.aws.glue.GlueCatalog} should use external lock or not, default to false.
+   * If set to true, it will use DynamoDB to enforce locking during commits.
+   */
+  public static final String GLUE_CATALOG_LOCK_ENABLED = "gluecatalog.lock.enabled";
+  public static final boolean GLUE_CATALOG_LOCK_ENABLED_DEFAULT = false;
+
+  /**
+   * The DynamoDB table used for locking.
+   * One lock table is designed to be used for only one catalog.
```

Comment:
Because it is not straightforward to get a consistent glue catalog ID, unless we always initialize a client to fetch the AWS account and region from the caller, which creates lots of complications. By doing this, you basically have one dynamo lock table for one Glue catalog, which is simpler.